core/src/main/java/tech/pegasys/eth2signer/core/service/jsonrpc/FcJsonRpc.java

```diff
@@ -107,34 +109,32 @@ public boolean filecoinWalletVerify(
       @JsonRpcParam("address") final String filecoinAddress,
       @JsonRpcParam("data") final String dataToVerify,
       @JsonRpcParam("signature") final FilecoinSignature filecoinSignature) {
-    final Optional<ArtifactSigner> signer = fcSigners.getSigner(filecoinAddress);
+    final FilecoinAddress address = FilecoinAddress.fromString(filecoinAddress);
+    final Bytes signature = Bytes.fromBase64String(filecoinSignature.getData());
 
-    if (signer.isPresent()) {
-      final Bytes bytesToVerify = Bytes.fromBase64String(dataToVerify);
-      final ArtifactSignature artifactSignature = createArtifactSignature(filecoinSignature);
-      return signer.get().verify(bytesToVerify, artifactSignature);
-    } else {
-      throw new FilecoinSignerNotFoundException();
+    switch (address.getProtocol()) {
+      case SECP256K1:
+        return FilecoinVerify.verify(
+            address, Bytes.fromBase64String(dataToVerify), createSecpArtifactSignature(signature));
+      case BLS:
+        return FilecoinVerify.verify(
+            address,
+            Bytes.fromBase64String(dataToVerify),
+            new BlsArtifactSignature(BLSSignature.fromBytes(signature)));
+      default:
+        throw new IllegalArgumentException("Invalid address protocol type");
     }
   }
 
-  private ArtifactSignature createArtifactSignature(final FilecoinSignature filecoinSignature) {
-    final Bytes signature = Bytes.fromBase64String(filecoinSignature.getData());
-    switch (filecoinSignature.getType()) {
-      case SECP_VALUE:
-        final Bytes r = signature.slice(0, 32);
-        final Bytes s = signature.slice(32, 32);
-        final Bytes v = signature.slice(64);
-        return new SecpArtifactSignature(
-            new Signature(
-                Numeric.toBigInt(v.toArrayUnsafe()),
-                Numeric.toBigInt(r.toArrayUnsafe()),
-                Numeric.toBigInt(s.toArrayUnsafe())));
-      case BLS_VALUE:
-        return new BlsArtifactSignature(BLSSignature.fromBytes(signature));
-      default:
-        throw new IllegalArgumentException("Invalid Signature type");
-    }
+  private SecpArtifactSignature createSecpArtifactSignature(final Bytes signature) {
```

Comment:
suspect this belongs somewhere other than the JsonRpc handling class.
In GENERAL, I like the idea of keeping this class super thin, and simply being entry points rather than doing work ... haven't been great on that so far though